### PR TITLE
Introducing JitPlugin

### PR DESCRIPTION
**Pull request by janeyx99**
This PR is step 1 to covering JIT'd methods and functions. Step 2 (using it in CI) is here: #56310.

1. This PR introduces a package `coverage_plugins` that hosts JITPlugin.
2. We also bring in a `.coveragerc` file that is used in CI to omit the files we don't want to report on (e.g., temporary directories or test or utils.)

**Disclaimer: This PR does NOT use the plug-in. Nothing should change as a result.**

Test:
CI. Coverage should not go down.

If you're interested in testing this plug-in locally, you should:
`pip install -e tools/coverage_plugins_package` from the root directory.
Add the following lines to `.coveragerc` under `[run]`
```
plugins =
    coverage_plugins.jit_plugin
```
And then try:
`coverage run test/test_jit.py TestAsync.test_async_script_no_script_mod`

You should see `.coverage.jit` show up at the end. You can then run `coverage combine --append` and `coverage debug data` to see that some files in `torch/jit` are covered.

### CODE REVIEW

File: tools/coverage_plugins_package/setup.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import setuptools
+
+with open("README.md", "r", encoding="utf-8") as fh:
+    long_description = fh.read()
+
+setuptools.setup(
+    name="coverage-plugins",
+    version="0.0.1",
+    author="Jane Xu",
```
Comment (janeyx99) [inline tools/coverage_plugins_package/setup.py:9]: Should this be PyTorch? And the email something made up?

File: tools/coverage_plugins_package/setup.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    author_email="janeyx@fb.com",
```
Comment (walterddr) [inline tools/coverage_plugins_package/setup.py:10]: make it same as top level setup.py?
```suggestion
    author='PyTorch Team',
    author_email='packages@pytorch.org',
```

File: tools/coverage_plugins_package/src/coverage_plugins/jit_plugin.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            obj = frame.f_locals['fn']
+            if is_not_builtin_class(obj) or ismodule(obj) or ismethod(obj) or isfunction(obj) or iscode(obj):
+                filename = getsourcefile(obj)
+                sourcelines, starting_lineno = getsourcelines(obj)
+                line_data = {filename: range(starting_lineno, starting_lineno + len(sourcelines))}
+                cov_data.add_lines(line_data)
+        super().dynamic_context(frame)
+
+def coverage_init(reg, options):
+    reg.add_dynamic_context(JitPlugin())
```
Comment (walterddr) [inline tools/coverage_plugins_package/src/coverage_plugins/jit_plugin.py:37]: since this is not being used yet. is it possible to add a simple test?

**Review by walterddr — APPROVED** [review]
looks good to me

File: tools/coverage_plugins_package/src/coverage_plugins/jit_plugin.py (hunk 3 above)
Comment (janeyx99) [reply tools/coverage_plugins_package/src/coverage_plugins/jit_plugin.py:37]: Hm true, I can try that.

**Comment by janeyx99** [comment]
I notice the windows test failures are my bad, so I'll fix those before landing as well

**Review by sab-regime — APPROVED** [review]

**Review by sab-regime — APPROVED** [review]

**Pull request closed without merging**